realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -57,6 +59,8 @@ JNIEXPORT void JNICALL Java_io_realm_internal_SharedRealm_nativeInit(JNIEnv* env
 {
     TR_ENTER()
 
+    hack_init();
```

Comment:
Shouldn't this be called in JNI_OnLoad or something? There is no guarantee that this code is the first native code being called?